c/src/lib/libbsp/arm/shared/stm32fxxxx/uart/hal-uart-interface.c

```diff
@@ -0,0 +1,499 @@
+ * found in the file LICENSE in this distribution or at
+ * http://www.rtems.com/license/LICENSE.
+ */
+
+//================== STMF32 Support Functions =================================
+#include <hal-uart-interface.h>
+#include <hal-utils.h>
+#include <stm32f-processor-specific.h>
+#include <bspopts.h>
+
+#include stm_processor_header(TARGET_STM_PROCESSOR_PREFIX)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, rcc)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, rcc_ex)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, gpio)
+
+// A set of UART handle used in the HAL code.  These handles are used
+// for both the termios console uarts as well as the non-console uarts.
+UART_HandleTypeDef UartHandles[NUM_PROCESSOR_CONSOLE_UARTS+NUM_PROCESSOR_NON_CONSOLE_UARTS];
+
+static stm32f_base_uart_driver_entry* stm32f_get_driver_entry_from_handle(
+  const UART_HandleTypeDef *huart
+)
+{
+  uint32_t i;
+
+  for ( i = 0UL; i < COUNTOF(stm32f_console_driver_table); i++ ) {
+    if ( stm32f_console_driver_table[i].base_driver_info.handle == huart ) {
+      return (stm32f_base_uart_driver_entry*) &(stm32f_console_driver_table[i].base_driver_info);
+    }
+  }
+
+  for ( i = 0UL; i < COUNTOF(stm32f_uart_driver_table); i++ ) {
+    if ( stm32f_uart_driver_table[i].base_driver_info.handle == huart ) {
+      return (stm32f_base_uart_driver_entry*) &(stm32f_uart_driver_table[i].base_driver_info);
+    }
+  }
+
+  return NULL;
+}
+
+stm32f_console_driver_entry* stm32f_get_console_driver_entry_from_handle(
+  const UART_HandleTypeDef *huart
+)
+{
+  uint32_t i;
+
+  for ( i = 0UL; i < COUNTOF(stm32f_console_driver_table); i++ ) {
+    if ( stm32f_console_driver_table[i].base_driver_info.handle == huart ) {
+      return (stm32f_console_driver_entry*) &(stm32f_console_driver_table[i]);
+    }
+  }
+
+  return NULL;
+}
+
+static void stm32f_init_uart_clock(
+  const stm32f_uart Uart
+)
+{
+  switch ( Uart )
+  {
+#if defined(STM32F7_ENABLE_USART_1)
+  case STM32F_UART1:
+    __HAL_RCC_USART1_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_2)
+  case STM32F_UART2:
+    __HAL_RCC_USART2_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_3)
+  case STM32F_UART3:
+    __HAL_RCC_USART3_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_4)
+  case STM32F_UART4:
+    __HAL_RCC_UART4_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_5)
+  case STM32F_UART5:
+    __HAL_RCC_UART5_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_6)
+  case STM32F_UART6:
+    __HAL_RCC_USART6_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_USART_7)
+    case STM32F_UART7:
+    __HAL_RCC_UART7_CLK_ENABLE();
+    break;
+#endif
+
+#if defined(STM32F7_ENABLE_UART_8)
+    case STM32F_UART8:
+    __HAL_RCC_UART8_CLK_ENABLE();
+    break;
+#endif
+  default:
+  case STM32F_INVALID_UART:
+    while(1) {
+      // you tried to reference a UART that is not enabled
```

Comment:
I will call stm32f_error_handler so that we can better define the behavior later.